### PR TITLE
Add Hugo layout to hide release information on outdated k8s.io versions

### LINKED ISSUE #29428: Create a Hugo layout to not display release information on older versions of k8s.io
## Problem

When a Kubernetes release happens, like 1.22, the now "old" site becomes a subdomain snapshot (ie: https://v1-21.docs.kubernetes.io/). This is great for archival purposes, however, if you navigate to the snapshot's release page (https://v1-21.docs.kubernetes.io/releases/) it is stuck at that point in time.

This problem exists because each older site is a branch. 1.21 is on the `release-1.21` branch.

## Solution

We should have a single source of truth (`main`) and all other subdomains should automatically point to the current documentation.

I would recommend adding a shortcode, much like we do to create the yellow box* on older pages. 

image

## Pointers

The yellow box shortcode exists on all versions and branches of k/website, but is contingent on a hugo configuration value (set [here](https://github.com/kubernetes/website/blob/main/config.toml#L147)). For example:

- 1.22 on the `main` branch does not have the box appear and the config.toml is set to false (https://github.com/kubernetes/website/blob/main/config.toml#L147).
- 1.21 on the `release` branch does have the box appear and the config.toml is set to true (https://github.com/kubernetes/website/blob/release-1.21/config.toml#L147)

The bit of code that actually generates the parameter is in the following layout partial:
- https://github.com/kubernetes/website/blob/main/layouts/partials/deprecation-warning.html
```
{{ if (.Site.Param "deprecated") }}
<section id="deprecation-warning">
  <div class="content deprecation-warning">
    <h3>
      {{ T "deprecation_title" }} {{ .Param "version" }}
    </h3>
    <p> Kubernetes {{ .Param "version" }} {{ T "deprecation_warning" }}
      <a href="{{ site.Params.currentUrl }}">{{ T "latest_version" }}</a>
    </p>
  </div>
</section>
{{ end }}
```

The "T" before the value references using the translated strings that are localized in the relative language files. For example, the English translation is located [here](https://github.com/kubernetes/website/blob/main/data/i18n/en/en.toml).

For more information on the releases page and how it's made, check out the PR: https://github.com/kubernetes/website/pull/27929

This is a well scoped issue, but maybe slightly more advanced for a first issue. I will mark it as such either way.

/sig docs
/good-first-issue

### PR DESCRIPTION
**Pull request by dipesh-rawat**
Resolves #29428

### CODE REVIEW

**Comment by dipesh-rawat** [comment]
/cc @sftim 
/cc @jimangel 

For review and feedback as requestors for this feature.

File: layouts/docs/release-info.html
Comment (sftim) [inline layouts/docs/release-info.html:1]: When is this layout used? What should display for a previous release?

File: data/i18n/en/en.toml [hunk 1]
```diff
@@ -337,6 +337,9 @@ other = "Options"
 [outdated_blog__message]
 other = "This article is more than one year old. Older articles may contain outdated content. Check that the information in the page has not become incorrect since its publication."
 
+[outdated_release_info_warning]
+other = " documentation is no longer actively maintained. The version you are currently viewing is a static snapshot. For the latest release information, please refer to the "
```
Comment (sftim) [inline data/i18n/en/en.toml:341]: Partial sentences are very hard to localize. Please consider splitting this up into something where the localized strings are whole sentences.

File: content/en/releases/_index.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 linktitle: Release History
 title: Releases
 type: docs
+layout: release-info
```
Comment (sftim) [inline content/en/releases/_index.md:5]: Should every page within `/releases` have this marker? Maybe it's not needed, as we have `if eq .Section "releases"` logic elsewhere.

**Review by sftim — COMMENTED** [review]
Thanks.

I don't think this is ready to merge. Please try this with a local change to `hugo.toml` and check to your own satisfaction that this works right, including for a non-English localization.

**Comment by dipesh-rawat** [comment]
I'm closing this pull request due to some messy git commits from my end. I've already opened PR #41370 where I will address the review comments provided. Apologies for the inconvenience caused.

/close

**Pull request closed without merging**